code.js

```diff
@@ -1,9 +1,26 @@
+
 function factorial(n) {
-    if(n === 0) return 1;
-    else return n * factorial(n - 1);
+    let result = 1;
+
+    if (n < 0) { return null; } // Negative Numbers Aren't Allowed
+    if (n == 0) { return 1; }
+
+    for (let i = 1; i <= n; i++) {
+        result *= i;
+    }
+
+    return result;
 }
 
 function e(n) {
-    if(n === 0) return 1;
-    else return 1.0 / factorial(n) + e(n - 1);
+    let result = 1.0;
+
+    if (n < 0) { return null; } // Negative Numbers Aren't Allowed
+    if (n == 0) { return result; }
+
+    for (let i = 1; i <= n; i++) {
+        result +=  (1.0 / factorial(i));
```

Comment:
Might there be a better way to do this? You're recomputing the factorial every time...